raiden/network/channelgraph.py

```diff
@@ -166,16 +169,18 @@ def get_best_routes(
         if channel.state != CHANNEL_STATE_OPENED:
             continue
 
-        network_state = nodeaddresses_statuses[partner_address]
         route_state = channel_to_routestate(channel, partner_address)
-
-        if network_state == NODE_NETWORK_REACHABLE:
-            online_nodes.append(route_state)
-
-        elif network_state == NODE_NETWORK_UNKNOWN:
-            unknown_nodes.append(route_state)
-
-    return online_nodes + unknown_nodes
+        routable_neighbors.append(
+            RouteOrderedItem(
+                hops,
+                nodeaddresses_statuses[partner_address],
```

Comment:
Until you do a transfer to the node, its state is considered 'unknown'. After that, a healthcheck() pings the node and updates the status.
We must either start healthcheck() by default for every neighbor, or not rely on the nodeaddresses_statuses at all. Otherwise it will be possible to have a scenario where the tx is (maliciously) routed through a more 'expensive' network (whatever 'expensive' means - be it length, total cost of tx, time...).

consider following topology

```
A +----> B ---+> C
  |           |
  +----> D ---+

```



A wants to do tx to C. Malicious actor B wants to increase his profit for relaying a transaction, so he will send a transfer to A (or to itself through A), thus making the AB channel monitored & in a 'reachable' state for node A. Honest actor D is also reachable, but at the moment he is not monitored and 'unknown' to A.
Without fix, the algorithm will always choose channel AB for the transaction, not taking into an account a cheaper route is available. 
LB: I think route algo should randomize targets if cost of routing is equal for multiple of them, to avoid someone hijacking the transfers (i.e. by generating address that puts them on the top of the internal routelist b/c of how strings are ordered)